subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -1038,13 +1038,16 @@ int lwm2m_send_message(struct lwm2m_message *msg)
 	}
 
 	if (msg->type == COAP_TYPE_CON) {
-		/* don't re-queue the retransmit work on retransmits */
-		if (msg->send_attempts > 1) {
-			return 0;
-		}
+		s32_t remaining = k_delayed_work_remaining_get(
+					&msg->ctx->retransmit_work);
 
-		k_delayed_work_submit(&msg->ctx->retransmit_work,
-				      msg->pending->timeout);
+		/* If the item is already pending and its timeout is smaller
+		 * than the new one, skip the submission.
+		 */
+		if (remaining == 0 || remaining > msg->pending->timeout) {
+			k_delayed_work_submit(&msg->ctx->retransmit_work,
+					      msg->pending->timeout);
```

Comment:
This will not work after we switch to k_timeout_t

Reply:
Fixed